l10n_br_sale_stock/models/sale_order.py

```diff
@@ -34,3 +34,29 @@ def _compute_get_button_create_invoice_invisible(self):
                     button_create_invoice_invisible = True
 
         self.button_create_invoice_invisible = button_create_invoice_invisible
+
+    @api.onchange("partner_shipping_id")
+    def _onchange_partner_shipping_id(self):
+        """
+        Caso ocorra a alteração do campo Endereço de Entrega/partner_shipping_id
+        depois do Pedido confirmado os stock.picking relacionados ficam com o
+        partner_id anterior, o que é errado, o metodo original apenas mostra uma
+        mensagem de alerta na tela orientando o usuário a corrigir manualmente,
+        mas como na localização com esse modulo pode se definir a criação da
+        Invoice a partir do stock.picking é melhor garantir a alteração do campo
+        partner_id da stock.picking afim de evitar erros na criação da Invoice.
+        :return: super()
+        """
+        # TODO: Verificar essa questão na migração a partir da v14
+
+        pickings = self.picking_ids.filtered(
+            lambda p: p.state not in ["done", "cancel"]
+            and p.partner_id != self.partner_shipping_id
+        )
+        for picking in pickings:
+            # Atribuição da forma abaixo por algum motivo não funciona
+            # apenas o write que está abaixo
+            # picking.partner_id = self.partner_shipping_id
+            picking.write({"partner_id": self.partner_shipping_id.id})
```

Comment:
Você não precisa fazer um for, você pode fazer simplesmente:

pickings.write({"partner_id": self.partner_shipping_id.id})